hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntitySelectFetchInitializer.java

```diff
@@ -186,6 +181,15 @@ public void initializeInstance(RowProcessingState rowProcessingState) {
 			}
 		}
 
+		LoadingProxyEntry loadingProxy = loadContexts.findLoadingProxy( entityKey );
+
+
+		if ( loadingProxy != null ) {
+			entityInstance = loadingProxy.getProxyTarget();
+			isInitialized = true;
+			return;
+		}
+
```

Comment:
It seems that this is unnecessary, because the proxy handling below should do the same AFAIU.

Reply:
this is needed to avoid the StackOverflow